http/events.go

```diff
@@ -174,10 +275,47 @@ func prependNamespacePatterns(patterns []string, requestNamespace *namespace.Nam
 	newPatterns := make([]string, 0, len(patterns)+1)
 	newPatterns = append(newPatterns, prepend)
 	for _, pattern := range patterns {
-		if strings.Trim(strings.TrimSpace(pattern), "/") == "" {
-			continue
+		if strings.Trim(pattern, "/") != "" {
+			newPatterns = append(newPatterns, path.Join(prepend, pattern))
 		}
-		newPatterns = append(newPatterns, path.Join(prepend, pattern, "/"))
 	}
 	return newPatterns
 }
+
+// validateSubscribeAccessLoop continually checks if the request has access to the subscribe endpoint in
+// its namespace. If the access check ever fails, then the cancel function is called and  the function returns.
+func validateSubscribeAccessLoop(core *vault.Core, ctx context.Context, cancel context.CancelFunc, req *logical.Request) {
+	// if something breaks, default to canceling the websocket
+	defer cancel()
+	for {
+		err := validateSubscribeAccess(core, ctx, req)
+		if err != nil {
+			return
+		}
+		// wait a while and try again, but quit the loop if the context finishes early
+		finished := func() bool {
+			ticker := time.NewTicker(webSocketRevalidationTime)
+			defer ticker.Stop()
+			select {
+			case <-ctx.Done():
+				return true
+			case <-ticker.C:
+				return false
+			}
+		}()
+		if finished {
+			return
+		}
+	}
+}
+
+// validateSubscribeAccess does a single check that the request has access to subscribe endpoint.
+func validateSubscribeAccess(core *vault.Core, ctx context.Context, req *logical.Request) error {
```

Comment:
This is perfectly fine but at the call site it looks like the error is being thrown away - would this be easier to read if this function was just inlined, as it's only a single function call and a log line anyway?

Reply:
It was initially more complicated, which is why it was a separate function. But I agree, it could probably be moved back inline now.